src/test/e2e/services/addon-service.e2e.test.ts

```diff
@@ -0,0 +1,74 @@
+import dbInit from '../helpers/database-init';
+import getLogger from '../../fixtures/no-logger';
+import { createTestConfig } from '../../config/test-config';
+import AddonService from '../../../lib/services/addon-service';
+import { IUnleashStores } from '../../../lib/types';
+
+import SimpleAddon from '../../../lib/services/addon-service-test-simple-addon';
+import TagTypeService from '../../../lib/services/tag-type-service';
+import { FEATURE_CREATED } from '../../../lib/types/events';
+
+const addonProvider = { simple: new SimpleAddon() };
+
+let db;
+let stores: IUnleashStores;
+let addonService: AddonService;
+
+beforeAll(async () => {
+    const config = createTestConfig({
+        server: { baseUriPath: '/test' },
+    });
+    db = await dbInit('addon_service_serial', getLogger);
+    stores = db.stores;
+    const tagTypeService = new TagTypeService(stores, config);
+    addonService = new AddonService(
+        stores,
+        config,
+        tagTypeService,
+        addonProvider,
+    );
+});
+
+afterAll(async () => {
+    if (db) {
+        await db.destroy();
+    }
+});
+afterEach(async () => {
+    const addons = await stores.addonStore.getAll();
+    const deleteAll = addons.map((a) => stores.addonStore.delete(a.id));
+    await Promise.all(deleteAll);
+});
+
+test('should only return active addons', async () => {
+    const config = {
+        provider: 'simple',
+        enabled: true,
+        parameters: {
+            url: 'http://localhost/wh',
+            var: 'some-value',
+        },
+        events: [FEATURE_CREATED],
+        description: '',
+    };
+
+    const config2 = {
+        provider: 'simple',
+        enabled: true,
+        parameters: {
+            url: 'http://localhost/wh',
+            var: 'some-value',
+        },
+        events: [FEATURE_CREATED],
+        description: '',
+    };
+
+    await addonService.createAddon(config, 'me@mail.com');
+    await addonService.createAddon(config2, 'me@mail.com');
+
+    const activeAddons = await addonService.fetchAddonConfigs();
+    const allAddons = await addonService.getAddons();
+
+    expect(activeAddons.length).toBe(1);
```

Comment:
The way this test is set up I would expect this to be 2. The fetchAddonConfigs function filters on enabled, and in this case both configs are set to true.

Reply:
You are absolute right. I will investigate.

Reply:
It was by chance, because of how things are loaded in the background. I have now improved the tests to have 3 addon configurations in total with two enabled and one disabled. 

In addition I have taken control over time to make sure the caching get expired correctly. By having multiple configs we ensure the test fails if one does not deal with the time aspect of things.